sources/api/netdog/src/main.rs

```diff
@@ -299,11 +327,27 @@ fn remove(args: &Args) -> Result<()> {
     Ok(())
 }
 
+/// Return the current IP address as JSON (intended for use as a settings generator)
+fn node_ip() -> Result<()> {
+    let ip =
+        fs::read_to_string(CURRENT_IP).context(error::CurrentIpReadFailed { path: CURRENT_IP })?;
+    // sundog expects JSON-serialized output
+    let output = serde_json::to_string(&ip).context(error::JsonSerialize { output: ip })?;
+    println!("{}", output);
+    Ok(())
+}
+
 fn run() -> Result<()> {
-    let args = parse_args(env::args())?;
-    match args.sub_command {
-        SubCommand::Install => install(&args)?,
-        SubCommand::Remove => remove(&args)?,
+    match parse_args(env::args())? {
+        (SubCommand::NodeIp, None) => node_ip()?,
+        (subcommand, Some(args)) => match subcommand {
+            SubCommand::Install => install(&args)?,
+            SubCommand::Remove => remove(&args)?,
+            SubCommand::NodeIp => {
+                usage_msg(format!("Subcommand 'node-ip' doesn't support arguments"))
+            }
+        },
+        (_, None) => usage_msg(format!("Subcommands 'install/remove' require arguments")),
```

Comment:
nit: this might be clearer overall, since it doesn't mix `NodeIp` in with the handling of `Install` and `Remove`:

```suggestion
    match parse_args(env::args())? {
        (SubCommand::NodeIp, None) => node_ip()?,
        (SubCommand::NodeIp, Some(_) => usage_msg(format!("Subcommand 'node-ip' doesn't support arguments")).
        (subcommand, Some(args)) => match subcommand {
            SubCommand::Install => install(&args)?,
            SubCommand::Remove => remove(&args)?,
        },
        (_, None) => usage_msg(format!("Subcommands 'install/remove' require arguments")),
```

Reply:
```
(_, None) => usage_msg(format!("Subcommands 'install/remove' require arguments"))
```

For this it might be better to use the `subcommand` parameter so you can add it to the format string.

Reply:
I suggest:

```rust
enum Subcommand {
    Install(Args),
    Remove(Args),
    NodeIp,
}
```

And

```rust
fn parse_args(args: env::Args) -> Result<SubCommand> {
```

Reply:
Actually since, since this would require us to stop using serde_plain to 'deserialize' the subcommand string, it's probably not worth it.

Reply:
@bcressey - The compiler wasn't smart enough to do your suggestion, it still wanted an exhaustive list of the enum variants inside the second match.  I opted to split out the options a bit further and just avoid a nested match altogether.